packages/edit-site/src/components/left-sidebar/navigation-panel/index.js

```diff
@@ -1,80 +1,101 @@
 /**
  * WordPress dependencies
  */
-import { useEffect, useRef } from '@wordpress/element';
+import { useEffect, useRef, useState } from '@wordpress/element';
 import {
 	__experimentalNavigation as Navigation,
 	__experimentalNavigationMenu as NavigationMenu,
+	__experimentalNavigationItem as NavigationItem,
 	__experimentalNavigationBackButton as NavigationBackButton,
+	createSlotFill,
 } from '@wordpress/components';
 import { useDispatch, useSelect } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
-import TemplateSwitcher from './template-switcher';
+import TemplatesMenu from './menus/templates';
+import TemplatePartsMenu from './menus/template-parts';
+
+export const {
+	Fill: NavigationPanelPreviewFill,
+	Slot: NavigationPanelPreviewSlot,
+} = createSlotFill( 'EditSiteNavigationPanelPreview' );
 
 const NavigationPanel = () => {
+	const [ activeMenu, setActiveMenu ] = useState( 'root' );
 	const ref = useRef();
 
 	useEffect( () => {
 		ref.current.focus();
 	}, [ ref ] );
 
-	const { templateId, templatePartId, templateType, page } = useSelect(
-		( select ) => {
-			const {
-				getTemplateId,
-				getTemplatePartId,
-				getTemplateType,
-				getPage,
-			} = select( 'core/edit-site' );
+	const { template, templatePart, templateType } = useSelect( ( select ) => {
+		const { getEntityRecord } = select( 'core' );
+		const { getTemplateId, getTemplatePartId, getTemplateType } = select(
+			'core/edit-site'
+		);
 
-			return {
-				templateId: getTemplateId(),
-				templatePartId: getTemplatePartId(),
-				templateType: getTemplateType(),
-				page: getPage(),
-			};
-		},
-		[]
-	);
+		const templateId = getTemplateId();
+		const templatePartId = getTemplatePartId();
+
+		return {
+			template: templateId
+				? getEntityRecord( 'postType', 'wp_template', templateId )
+				: null,
+			templatePart: templatePartId
+				? getEntityRecord(
+						'postType',
+						'wp_template_part',
+						templatePartId
+				  )
+				: null,
+			templateType: getTemplateType(),
+		};
+	}, [] );
 
-	const {
-		setTemplate,
-		addTemplate,
-		removeTemplate,
-		setTemplatePart,
-	} = useDispatch( 'core/edit-site' );
+	const { setTemplate, setTemplatePart } = useDispatch( 'core/edit-site' );
 
 	return (
 		<div className="edit-site-navigation-panel">
 			<Navigation
 				activeItem={
 					'wp_template' === templateType
-						? `template-${ templateId }`
-						: `template-part-${ templatePartId }`
+						? `template-${ template?.slug }`
+						: `template-part-${ templatePart?.slug }`
 				}
+				onActivateMenu={ setActiveMenu }
 			>
-				<NavigationBackButton
-					backButtonLabel={ __( 'Dashboard' ) }
-					className="edit-site-navigation-panel__back-to-dashboard"
-					href="index.php"
-					ref={ ref }
-				/>
+				{ activeMenu === 'root' && (
```

Comment:
I'm cool with this change, but I wonder if it might have a11y implications — either way I mean. 🤔

Reply:
Two back button is strange, that was the main reason I removed this back button.

Reply:
Right, two back buttons is strange, but at the same time one of them (back to dashboard) is "outside" the navigation menu context.
So, it might not be clear for visually impaired users why that button disappeared while navigating through the menu.

This is all anecdotal, though.
As I said, the change looks good to me, and it's super simple to revert. 🙂